src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigWatcher.java

```diff
@@ -45,6 +54,20 @@ public void onClose(KubernetesClientException e) {
     }
   }
 
+  @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
+  public void onInitialBuildConfigs(BuildConfigList buildConfigs) {
```

Comment:
It'd be nice if the kubernetes client could deal with that automatically - a listAndWatch() method that automatically filtered out old or duplicates